e2e-test/tsps_e2e_test.py

```diff
@@ -0,0 +1,395 @@
+from workspace_helper import create_workspace, delete_workspace
+from app_helper import create_app, poll_for_app_url
+import argparse
+import requests
+import os
+import json
+import uuid
+import time
+import logging
+
+
+# Upload data to WDS using APIs
+def upload_wds_data_using_api(wds_url, workspace_id, tsv_file_name, record_name):
+    #open TSV file in read mode
+    with open(tsv_file_name) as tsv_file:
+        request_file = tsv_file.read()
+
+    uri = f"{wds_url}/{workspace_id}/tsv/v0.2/{record_name}"
+    headers = {"Authorization": f"Bearer {azure_user_token}"}
+
+    response = requests.post(uri, files={'records':request_file}, headers=headers)
+
+    status_code = response.status_code
+
+    if status_code != 200:
+        raise Exception(response.text)
+
+    logging.info(f"Successfully uploaded data to WDS. Response: {response.json()}")
+
+
+# Create imputation method in CBAS
+def create_cbas_method(cbas_url, workspace_id, github_url, token):
+    method_name = "ImputationBeagle"
+    request_body = {
+        "method_name": method_name,
+        "method_source": "GitHub",
+        "method_url": f"{github_url}",
+        "method_version": "1"
+    }
+
+    uri = f"{cbas_url}/api/batch/v1/methods"
+    headers = {
+        "Authorization": f"Bearer {token}",
+        "accept": "application/json",
+        "Content-Type": "application/json"
+    }
+
+    response = requests.post(uri, json=request_body, headers=headers)
+    status_code = response.status_code
+
+    if status_code != 200:
+        raise Exception(response.text)
+
+    logging.info(f"Successfully created method {method_name} for workspace {workspace_id}")
+    response = json.loads(response.text)
+
+    return response['method_id']
+
+
+def share_workspace(orch_url, billing_project_name, workspace_name, email_to_share, owner_token):
```

Comment:
can we either add "owner" to the function name or add the access level as an input?